Update ch15-03-drop.md (#2049)
rust-lang/book: src/ch15-03-drop.md

`Drop` trait is that it’s taken care of automatically. Occasionally, however,
you might want to clean up a value early. One example is when using smart
pointers that manage locks: you might want to force the `drop` method that

> releases the lock to run some other code in the same scope can acquire the lock.

Reviewer: ```suggestion
releases the lock so that other code in the same scope can acquire the lock.
```
